core/validation/specific_validation/pathfinders/round_1.py

```diff
@@ -186,3 +188,38 @@ def _check_credible_plan_fields(extracted_table_dfs: dict[str, pd.DataFrame]) ->
                         )
                     )
     return error_messages
+
+
+def _check_actual_forecast_reporting_period(extracted_table_dfs: dict[str, pd.DataFrame]) -> list[Message]:
+    reporting_period = extracted_table_dfs["Reporting period"].iloc[0, 0]
+    submission_reporting_period_start_date = PF_REPORTING_PERIOD_TO_DATES_2[reporting_period]["start"]
+    pfcs_df = extracted_table_dfs["Project finance changes"]
```

Comment:
is this specifically for PFC's tab? Do you know if we also do this validation (Actual/forecast) for the other tabs such as Funding / Outcomes? I'm not 100% sure if we even do need to validate these other tabs, but if we do, is there a separate ticket, or should we reuse this func to do it?

Reply:
Yes specifically for PFCs - no strong need to validate for "Outputs", "Outcomes" and "Forecast and actual spend" tables as "Actual" versus "Forecast" in the column headers is not specified by the user, rather hard-coded. Here we require validation because the user selects "Actual", "Forecast" or "Cancelled" for "Actual, forecast or cancelled" field, and then selects an available reporting period for "Reporting period change takes place", and we need to make sure those align (the spreadsheet does not do this for us)

Reply:
Thinking about this some more, because the "Actual" and "Forecast" values in the "Outputs", "Outcomes" and "Forecast and actual spend" tables actually form part of the column headers, they are effectively validated already - e.g., if the user hypothetically modified the value "Forecast" for "Financial year 2024/25, (Apr to Jun)" to "Actual", then the calculated column header expected for the template "Financial year 2024/25, (Apr to Jun), Forecast" would not appear (as we would end up with "Financial year 2024/25, (Apr to Jun), Actual" instead, and the validation would fail at `_check_columns` (see `TableValidator` class)